code/controllers/subsystems/ghostroles.dm

```diff
@@ -177,3 +185,25 @@
 			for(var/i in S.spawnpoints)
 				update_spawnpoint_status_by_identifier(i)
 	return
+
+/datum/controller/subsystem/ghostroles/proc/add_spawn_atom(var/ghost_role_name, var/atom/spawn_atom)
+	if(ghost_role_name && spawn_atom)
+		var/datum/ghostspawner/G = spawners[ghost_role_name]
+		if(G)
+			G.spawn_atoms += spawn_atom
+			if(length(G.spawn_atoms) == 1)
+				G.enable()
+
+/datum/controller/subsystem/ghostroles/proc/remove_spawn_atom(var/ghost_role_name, var/atom/spawn_atom)
+	if(ghost_role_name && spawn_atom)
+		var/datum/ghostspawner/G = spawners[ghost_role_name]
+		if(G)
+			G.spawn_atoms -= spawn_atom
+			if(!length(G.spawn_atoms))
+				G.disable()
+
+/datum/controller/subsystem/ghostroles/proc/get_spawn_atoms(var/ghost_role_name)
```

Comment:
Is this function even used anywhere?

Reply:
nah, but is it bad to have it? i can't think of a reason to use it rn, but it could come in handy, like for an object or wizard spell that destroys all summoning atoms in the future